src/codegen/mod.rs

```diff
@@ -2198,7 +2198,13 @@ impl CodeGenerator for CompInfo {
             let prefix = ctx.trait_prefix();
             result.push(quote! {
                 impl #generics Default for #ty_for_impl {
-                    fn default() -> Self { unsafe { ::#prefix::mem::zeroed() } }
+                    fn default() -> Self {
+                        let mut s = ::#prefix::mem::MaybeUninit::<Self>::uninit();
+                        unsafe {
```

Comment:
This needs to check `ctx.options().rust_features().maybe_uninit`, to decide whether to use `mem::uninitialized()` or `MaybeUninit`. I guess we should at some point phase out support for some older rust versions, but since we have checks for this already let's use them.